File: rust/muse/src/algorithms/sorting/bubble_sort.rs
pub fn sort(arr: &mut [i32]) {
    let mut unsorted = arr.len();
    while unsorted > 1 {
        let mut margin = 0;
        for i in 1..unsorted {
            if arr[i - 1] > arr[i] {
                (arr[i - 1], arr[i]) = (arr[i], arr[i - 1]);
                margin = i;
            }
        }
        unsorted = margin;
    }
}

#[cfg(test)]
mod tests;

File: python/muse/algorithms/sorting/bubble_sort.py
def sort(arr: list[int]) -> None:
    unsorted = len(arr)
    while unsorted > 1:
        margin = 0
        for i in range(1, unsorted):
            if arr[i - 1] > arr[i]:
                arr[i - 1], arr[i] = arr[i], arr[i - 1]
                margin = i
        unsorted = margin
